From bd43b91b5a13c3089fcca01dd55c8e7ac7f811d0 Mon Sep 17 00:00:00 2001
From: David Gobbi <david.gobbi@gmail.com>
Date: Tue, 2 Jul 2019 20:41:49 -0600
Subject: [PATCH 3/3] Remove dead vtkImageViewer code from gdcmviewer.

The vtkImageViewer2 is always used, so there is no reason for the
older vtkImageViewer code to be kept.  Also, the dead code was
using vtkInstantiatorNewMacro, which is deprecated from VTK 8.1
onwards.
---
 Utilities/VTK/Applications/gdcmviewer.cxx | 35 +----------------------
 1 file changed, 1 insertion(+), 34 deletions(-)

diff --git a/Utilities/VTK/Applications/gdcmviewer.cxx b/Utilities/VTK/Applications/gdcmviewer.cxx
index c2d8c3c1c..7bd8014fe 100644
--- a/Utilities/VTK/Applications/gdcmviewer.cxx
+++ b/Utilities/VTK/Applications/gdcmviewer.cxx
@@ -31,7 +31,7 @@
 #include "vtkPolyDataMapper2D.h"
 #include "vtkImageReslice.h"
 #include "vtkRenderWindowInteractor.h"
-#include "vtkImageViewer.h"
+#include "vtkRenderWindow.h"
 #include "vtkPointData.h"
 #include "vtkImageMapToColors.h"
 #include "vtkLookupTable.h"
@@ -93,39 +93,6 @@ class vtkAngleWidget;
 #define vtkFloatingPointType double
 #endif
 
-//----------------------------------------------------------------------------
-// vtkImageViewer2 new interface wants SetSlice, but vtkImageViewer does not have
-// this new interface (what a pain), so let's fake a new interface to
-// vtkImageViewer without patching VTK
-class vtkGDCMImageViewer : public vtkImageViewer
-{
-public:
-  vtkTypeMacro(vtkGDCMImageViewer,vtkImageViewer);
-
-  static vtkGDCMImageViewer *New()
-    {
-#ifdef VTK_DEBUG_LEAKS
-    vtkDebugLeaks::ConstructClass("vtkGDCMImageViewer");
-#endif
-    return new vtkGDCMImageViewer;
-    }
-  int GetSlice() { return this->GetZSlice(); }
-  void SetSlice(int s) { this->SetZSlice(s); }
-
-  int GetSliceMin() { return this->GetWholeZMin(); }
-  int GetSliceMax() { return this->GetWholeZMax(); }
-#if VTK_MAJOR_VERSION >= 5
-  // TODO:
-  void AddInputConnection(vtkAlgorithmOutput* input) {(void)input;}
-#else
-  void AddInput(vtkImageData * input) {(void)input;}
-#endif
-  double GetOverlayVisibility() { return 0; }
-  void SetOverlayVisibility(double vis) {(void)vis;}
-};
-//vtkCxxRevisionMacro(vtkGDCMImageViewer, "$Revision: 1.30 $")
-vtkInstantiatorNewMacro(vtkGDCMImageViewer)
-
 #if VTK_MAJOR_VERSION >= 5
 #else
 class vtkImageColorViewer : public vtkImageViewer2
-- 
2.17.1

